aiida/backends/tests/export_and_import.py

```diff
@@ -697,6 +697,72 @@ def test_7(self):
             # Deleting the created temporary folder
             shutil.rmtree(temp_folder, ignore_errors=True)
 
+    def test_group_export(self):
+        """
+        Test that when exporting just a group, its nodes are also exported
+        """
+        import os
+        import shutil
+        import tempfile
+
+        from aiida.orm import load_node
+        from aiida.orm.calculation.job import JobCalculation
+        from aiida.orm.data.structure import StructureData
+        from aiida.orm.importexport import export
+        from aiida.common.datastructures import calc_states
+        from aiida.orm.querybuilder import QueryBuilder
+
+        # Creating a folder for the import/export files
+        temp_folder = tempfile.mkdtemp()
+        try:
+            # Create another user
+            new_email = "newuser@new.n"
+            user = self.backend.users.create(email=new_email)
+            user.store()
+
+            # Create a structure data node that has a calculation as output
+            sd1 = StructureData()
+            sd1.set_user(user)
+            sd1.label = 'sd1'
+            sd1.store()
+
+            jc1 = JobCalculation()
+            jc1.set_computer(self.computer)
+            jc1.set_option('resources', {"num_machines": 1, "num_mpiprocs_per_machine": 1})
+            jc1.set_user(user)
+            jc1.label = 'jc1'
+            jc1.store()
+            jc1.add_link_from(sd1)
+            jc1._set_state(calc_states.PARSING)
+
+            # Create a group and add the data inside
+            from aiida.orm.group import Group
+            g1 = Group(name="node_group")
+            g1.store()
+            g1.add_nodes([sd1, jc1])
+            g1_uuid = g1.uuid
+
+            # At this point we export the generated data
+            filename1 = os.path.join(temp_folder, "export1.tar.gz")
+            export([g1, jc1], outfile=filename1, silent=True)
+            n_uuids = [sd1.uuid, jc1.uuid]
+            self.clean_db()
+            self.insert_data()
+            import_data(filename1, silent=True)
+
+            # Check that the imported nodes are correctly imported and that
+            # the user assigned to the nodes is the right one
+            for uuid in n_uuids:
```

Comment:
If I was true on the above (that sd1 was there because of jc1) and all the tests pass now because you added the commented out code, please proceed. I will approve it